pig-runtime/test/org/partiql/pig/runtime/IntermediateRecordTests.kt

```diff
@@ -53,33 +58,66 @@ class IntermediateRecordTests {
         // Optional field that's not present
         ir.processOptionalField("baz") { foundFields.add(it) }
 
+        // Variadic field with minimum arity of 1
+        ir.processVariadicField("variadic_foo", 1) { foundFields.add(it) }
+
+        // Variadic field with minimum arity of 0 that is present with no values
+        // (should be removed so that the call to malformedIfAnyUnprocessedFieldsRemain does not throw)
+        ir.processVariadicField("variadic_empty", 0) { foundFields.add(it) }
+
+        // Variadic field with minimum arity of 0 that is not present (should not throw)
+        ir.processVariadicField("variadic_not_present", 0) { fail("should not get called")}
+
         assertDoesNotThrow { ir.malformedIfAnyUnprocessedFieldsRemain() }
-        assertEquals(listOf(ionInt(1), ionInt(2), ionInt(3)), foundFields)
+        assertEquals(listOf(ionInt(1), ionInt(2), ionInt(3), ionInt(4)), foundFields)
     }
 
-    @Test
-    fun requiredFieldMissing() {
-        val ir = createIntermediateRecord(mapOf("foo" to ionInt(1).asAnyElement()))
+    @ParameterizedTest
+    @MethodSource("parametersForMalformedTest")
+    fun malformedTests(tc: MalformedTestCase) {
         val ex = assertThrows<MalformedDomainDataException> {
-            ir.processRequiredField("bad_field") { error("should not be invoked") }
+            tc.blockThrowingMalformedDomainDataException()
+        }
+        tc.messageMustContainStrings.forEach {
+            assertTrue(ex.message!!.contains(it), "exception message must contain '$it'")
```

Comment:
nit: Hmm is there a better way to check the exception messages? Perhaps something similar to what is done in [TypeDomainParserErrorsTest.kt](https://github.com/partiql/partiql-ir-generator/blob/master/pig/test/org/partiql/pig/domain/TypeDomainParserErrorsTest.kt#L41)?

Reply:
I'd agree that would be better but that part of the code isn't currently factored to do that and doing so should be a separate effort.